### ISSUE TITLE
[2020 arXiv] Are we done with ImageNet?

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#78]
ImageNet validation dataset의 label noise를 정제하여 Reassessed Labels `ReaL` dataset을 만든 논문

## 1. What is wrong with ImageNet labels?
<img width="1393" alt="image" src="https://user-images.githubusercontent.com/49643709/177166310-f5340736-f44f-42a4-857b-c42427d0e329.png">

- `Single label per image` (top row of Figure 2) : real-world 이미지들은 다양한 object를 포함하고 있는데 ImageNet annotation은 각 이미지당 single label로 한정됨
- `Overly restrictive label proposals` (middle row of Figure 2) : 인터넷에 해당 class의 이미지들을 검색하여 사람에게 맞는 class인지 확인하는 collection procedure의 bias로 인해 plausible label을 가지게 됨
- `Arbitrary class distinctions` (bottom row of Figure 2) : 몇몇 ImageNet class는 semantically, visually indistinguishable함

### 2.1. Collecting a comprehensive set of proposals
<img width="975" alt="image" src="https://user-images.githubusercontent.com/49643709/177167887-2cc615c5-b360-453f-889c-c36b3b10c9f3.png">

ImageNet validation set에 label-noise가 있다고 하더라도 사람이 일일히 labeling하는 것은 비용이 많이 듬

이러한 문제를 해결하기 위해 19가지 모델을 이용하여 다음과 같은 방법을 이용하여 label proposals를 생성

```
1. 각 모델마다 각 이미지에 대해 logits와 probabilities 계산
    - ImageNet validation dataset은 총 50,000장임
    - 1개의 모델은 1개의 이미지에 대해 1000개 class에 대한 logit과 probability를 가짐
    - 즉 1개의 모델은 50M logits, 50M probabilities를 가짐
2. 각 모델마다 구한 logits, probabilities 중 largest 150k만 선택하여 image-label pairs를 생성
    - 1개의 모델은 largest 150k logits, largest 150k probabilities를 가짐
    - 선택된 largest 150k logits, 150k probabilities를 이용하여 image-label pairs를 생성
    - 즉 1개의 모델은 logit을 통해 150k의 image-label pairs와 probability를 통해 150k의 image-label pairs를 가짐
3. 모든 모델에 대하여 pool
    - 모든 모델에 대하여 image-label pairs를 pool
    - collections.Counter 개념으로 보면 이해하는데 있어서 수월할 듯
4. 1개의 모델에서 1번만 등장한 image-label pairs를 제거
    - image-label pairs 중에서 한번만 등장한 경우를 제거한다는 의미와 동일
5. 모든 이미지들에 대해 각 모델별 top-1 prediction image-label pairs와 original ILSCRV-2012 image-label pairs를 추가
```

이렇게 구한 label proposals의 수가 많아, 이를 줄이기 위해 저자들은 near perfect label coverage를 가지는 model의 subset를 다음과 같은 방법을 통해 찾음

```
1. Gold standard를 생성
    - gold standard : 5명의 vision experts가 256장의 이미지를 라벨링한 데이터셋임
2. Model 선택
    - 위에서 각 모델별 label proposals가 존재
    - 이를 이용하여 gold standard에 대해 precision & recall을 구함
    - recall을 97% 이상 유지하면서 highest precision을 가지는 6개의 모델을 선택
```

6개의 모델들을 이용하여 똑같은 과정 (19개의 모델에 대하여 label proposals를 구한 과정)을 진행하여 label proposals를 generate

저자들은 모델의 수를 줄임으로서 평균적으로 이미지당 13개의 label에서 7.4개로 줄일 수 있었다고 함

### 2.2. Human evaluation of proposed labels
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/49643709/177178083-42c37eea-bbe5-4a2c-8982-53afb814fe2b.png">

전체 validation set에 대해서 candidate label를 새로 만들었기에, 어떤 이미지들이 사람에게 평가받아야할 지에 대해 고민해야함

모든 모델들이 original ImageNet label과 똑같이 동의한다면, 굳이 볼 필요가 없다고 볼 수 있으며 이를 통해 50,000장에서 24,889장으로 라벨링할 이미지 수를 줄일 수 있었다고 함

8가지 이상의 label proposals를 가지고 있는 이미지에 대해 WordNet hierarchy에 따라 multiple labeling tasks로 변경하여 37,988장을 라벨링해야됨
(모델이 생각했을 때 이미지에 다양한 object가 있다면, WordNet hierarchy에 따라 multiple labeling task로 나눈다는 의미)

해당 작업은 crowdsourcing platform을 이용하여 5명의 human annotators가 해당 작업을 수행함
(single image는 최대 8개의 candidate labels를 가질 수 있으며, candidate labels가 이미지 내에 존재하는지를 체크)

### 2.3. From human ratings to labels and a metric

Human annotators가 수행한 label을 combine한 후, 모든 proposed label에 대해 유지할지 버릴지에 대한 single binary decision을 수행

동물 class의 경우, uncertainty가 많으며 expert opinion이 필요하기에 ImageNet label이 동물인 이미지에 대하여 ImageNet label을 추가

저자들은 EM 알고리즘의 일종인 `Dawid-Skene Algorithm`을 사용하여 single binary decision을 정하였으며, 이는 majority voting보다 효과적이었다고 함

```
Dawid-Skene Algorithm

1. Using the labels given by human annotators, estimate the most likely "correct" label for each image
2. Based on the estimated correct answer for each image, compute the error rates for each worker
3. Considering the error rates for each worker, recompute the most likely "correct" label for each object
4. Go to step 2
```

이를 통해 46,837 이미지에 대해 57,553 label을 붙였으며, 3,163 이미지들은 아무런 label을 할당받지 못하였기에 버렸다고 함

저자들은 새로운 validation set labels에 대해 ReaL accuracy라는 metric을 제안함
(model의 top-1 prediction이 label set에 포함되면 correct, 아니면 incorrect)

해당 metric은 model이 이미지 내에 다양한 class의 object가 있더라도 penalise하지 않기에 좋다고 함

## 3. Re-evaluating the state of the art
<img width="450" alt="image" src="https://user-images.githubusercontent.com/49643709/177188098-84c6c7fc-a687-406f-8f24-fbfc416ef058.png">

저자들은 기존의 모델들에 대하여 ReaL accuracy를 측정함

초기 모델들의 경우, ImageNet accuracy와 ReaL accuracy는 strong linear relationship을 가지고 있음 (slope = 0.86)

최근의 모델들에서도 ImageNet accuracy와 ReaL accuracy는 strong linear relationship을 가지고 있는데 기울기가 감소했음 (slope = 0.51)

또한 몇몇 모델들의 경우 original ImageNet labels보다 ReaL accuracy가 더 좋음
(Big transfer (BiT): General visual representation learning, Self-training with noisy student improves imagenet classification)

이를 통해 저자들은 ImageNet evaluation metric으로서 기능을 다 했다고 주장함

이를 보다 확실히하기 위해, 저자들은 ReaL에서는 맞췄는데 ImageNet label에서 틀렸다고 하는 모든 이미지들을 모아서 사람은 어떤 label을 더 prefer하는지를 실험

<img width="450" alt="image" src="https://user-images.githubusercontent.com/49643709/177190106-646613ae-43bf-4f56-964c-945d22812657.png">

초기 모델들의 경우 human preference를 예측하는데 ImageNet label보다 못했으나, 최근 모델들은 ImageNet label보다 human preference가 높음

### 3.1. Beyond single label predictions

기존의 ReaL accuracy는 model의 top-1 predection이 label set 안에 들어오기만 하면 됐음

이에 더 나아가서 model의 top predictions가 label set 안에 들어오는지를 더 strict한 metric으로 제안함

기존의 metric이 near-perfect accuracy를 보인다면, 사용하기에 적합한 metric이 될 것임

### 3.2. An analysis of co-occuring classes 
<img width="450" alt="image" src="https://user-images.githubusercontent.com/49643709/177260201-8229ef8f-a388-485e-bec2-df9f52a8eff0.png">

저자들은 ImageNet 데이터를 ReaL label로 다시 annotation을 하면서, 약 29%는 multiple synonym labels를 가지는 것을 관측함
(sunglass & sunglasses, bathtub & tub, promontory & cliff, laptop & notebook, keyboard & desk, cucumber & zucchini, hammer & nail, ...)

저자들은 이러한 현상을 보고, ImageNet 모델들은 해당 이미지에서 어떻게 작동하는지에 대한 의문을 가짐
(맞는 labels 중에서 하나를 random하게 고를지 아니면 labeling procedure에서의 bias를 배워서 맞추는지)

ReaL labels 중에서 random uniform하게 label을 predict한 결과에 대해 expected per-class accuracy를 compute한 경우를 `unbiased oracle model`이라고 명칭

저자들은 다음과 같은 과정을 통해 253개의 class의 ambiguous class pair 혹은 frequently appear class pair를 얻음

```
1. 이미지의 ReaL labels에 ImageNet label이 있는 모든 케이스를 모음
2. `Unbiased oracle`이 90%보다 낮은 accuracy를 보이는 class만 선택
3. Fine-grained animal class의 경우 labeling noise가 있을 수 있기에 제외
```

만약 ImageNet labeling procedure에 bias가 없다면, 모델의 per-class accuracy는 `unbiased oracle model`과 같을 것임

<img width="1101" alt="image" src="https://user-images.githubusercontent.com/49643709/177261318-280a5e10-2953-4a88-832b-dbd3aef75ca9.png">

허나 저자들은 몇몇 사례들을 분석하며 bias가 있음을 보임

예를 들어 75%가 넘는 이미지에 대해 desktop computer는 monitor, computer keyboard, desk와 같이 등장함

만약 label bias가 없다면 이는 약 30%의 per-class accuracy를 보여야하는데 ImageNet으로 학습한 모델들은 그렇지 않았음

이는 보편적인 현상인지에 대하여 19가지 모델들에 대하여 실험한 결과는 다음과 같음

<img width="450" alt="image" src="https://user-images.githubusercontent.com/49643709/177261007-7982f6cd-a0b2-434c-a5c4-100f481715f7.png">

저자들은 해당 현상을 통해 ImageNet의 성능 발전이 ImageNet의 labeling process bias를 학습하여 올라간 것이며, 이러한 이유로 인해 ImageNet에서의 성능 gain이 ReaL accuracy에서 partially transfer된다고 주장함

## 4. Analyzing the reamining mistakes

가장 성능이 좋은 모델은 ImageNet에서 11%, ReaL에서 9% error rate를 보임

저자들은 어떠한 이유로 인해 이러한 error가 발생하는지를 알기 위해 실험을 진행함

ImageNet과 ReaL에서 실수한 경우에 대해, model의 prediction과 실제 label을 사람에게 보여주어 모델이 잘못한건지, 아니면 label이 incorrect 혹은 incomplete한 것인지, 아니면 label이 fine-grained class라서 구분하기 어려운지를 체크하였으며 결과는 다음과 같음

분석의 편의를 더하기 위해 ImageNet accuracy를 기준으로 sorting

<img width="500" alt="image" src="https://user-images.githubusercontent.com/49643709/177264134-086ec230-9429-4680-ba7a-ed3df7d79dd7.png">

- Figure 7 top row : clear mistake의 수는 확연히 감소한 것을 확인할 수 있음
- Figure 7 middle row : 사람이 봤을 때 model prediction이 맞다고 생각되는 경우가 존재하며, ReaL보다 ImageNet에서 더 비율이 많음
- Figure 7 bottom row : non-expert가 봤을 떄, 구분이 힘든 경우가 존재

이를 통해 ReaL label이 ImageNet과 같이 fine-grained classes의 quality를 잘 보존하며, 나머지에 있어서 ImageNet에 비해 noise가 매우 적은 유의미한 benchmark임을 보임

## 5. Improving ImageNet training

저자들은 dataset이 noisy한 경우에 어떻게 극복할 수 있는지 2가지 방법을 제안함

1. Training objective를 multiple non-exclusive prediction을 할 수 있게 변경
2. 사람이 봤을 때 ImageNet label보다 더 신뢰되는 model을 이용하여 ImageNet label noise filtering

구체적으로 기존의 softmax cross-entropy loss 대신 sigmoid cross-entropy loss를 이용하여 학습하며, BiT-L을 이용하여 ImageNet label noise를 cleansing함

실험 결과는 아래와 같음

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/49643709/177266844-98675416-4122-40b5-a604-faaa5258d2e8.png">
<img width="1174" alt="image" src="https://user-images.githubusercontent.com/49643709/177267916-86e63d88-40a3-406b-9a1d-7ad75fd47e0e.png">
<img width="1099" alt="image" src="https://user-images.githubusercontent.com/49643709/177268010-09b2b30f-0700-4da7-9516-8ab9e0019235.png">

저자들은 실험 결과에 대해 다음과 같이 분석함

1. Training on clean ImageNet data consistently improves accuracy of the resulting model
2. Changing the softmax loss to the sigmoid loss also results in consistent accuracy improvements across all ResNet architectures and training settings
3. Combining clean data and sigmoid loss leads to further improvements
4. Opposed to the standard setting (softmax loss, full ImageNet data) which overfits for long schedules, training for longer using clean labels and sigmoid loss results in large gains
